src/components/network-map-tab.js

```diff
@@ -0,0 +1,351 @@
+        return (
+            <Menu
+                id={equipmentMenu.equipment.id}
+                position={[
+                    equipmentMenu.position[0],
+                    equipmentMenu.position[1],
+                ]}
+                handleClose={closeEquipmentMenu}
+                handleViewInSpreadsheet={handleViewInSpreadsheet}
+                {...props}
+            />
+        );
+    }
+
+    const MenuLine = withLineMenu(BaseEquipmentMenu);
+
+    const MenuSubstation = withEquipmentMenu(
+        BaseEquipmentMenu,
+        'substation-menu',
+        equipments.substations
+    );
+
+    const MenuVoltageLevel = withEquipmentMenu(
+        BaseEquipmentMenu,
+        'voltage-level-menu',
+        equipments.voltageLevels
+    );
+
+    function showEquipmentMenu(equipment, x, y, type) {
+        setEquipmentMenu({
+            position: [x, y],
+            equipment: equipment,
+            equipmentType: type,
+            display: true,
+        });
+    }
+
+    function closeEquipmentMenu() {
+        setEquipmentMenu({
+            display: false,
+        });
+    }
+
+    function handleViewInSpreadsheet(equipmentType, equipmentId) {
+        showInSpreadsheet({
+            equipmentType: equipmentType,
+            equipmentId: equipmentId,
+        });
+        closeEquipmentMenu();
+    }
+
+    function closeChoiceVoltageLevelMenu() {
+        setChoiceVoltageLevelsSubstationId(null);
+    }
+
+    function choiceVoltageLevel(voltageLevelId) {
+        openVoltageLevel(voltageLevelId);
+        closeChoiceVoltageLevelMenu();
+    }
+
+    const voltageLevelMenuClick = (equipment, x, y) => {
+        showEquipmentMenu(equipment, x, y, equipments.voltageLevels);
+    };
+
+    useEffect(() => {
+        console.info(`Loading geo data of study '${studyUuid}'...`);
+
+        const substationPositions = fetchSubstationPositions(studyUuid);
+        const linePositions = fetchLinePositions(studyUuid);
+        setWaitingLoadGeoData(true);
+
+        Promise.all([substationPositions, linePositions])
+            .then((values) => {
+                const geoData = new GeoData();
+                geoData.setSubstationPositions(values[0]);
+                geoData.setLinePositions(values[1]);
+                setGeoData(geoData);
+                setWaitingLoadGeoData(false);
+            })
+            .catch(function (error) {
+                console.error(error.message);
+                setWaitingLoadGeoData(false);
+                setErrorMessage('geoDataLoadingFail');
+            });
+        // Note: studyUuid and dispatch don't change
+    }, [studyUuid, setWaitingLoadGeoData, setErrorMessage, setGeoData]);
+
+    const chooseVoltageLevelForSubstation = useCallback(
+        (idSubstation, x, y) => {
+            setChoiceVoltageLevelsSubstationId(idSubstation);
+            setPosition([x, y]);
+        },
+        []
+    );
+
+    let choiceVoltageLevelsSubstation = null;
```

Comment:
What is this code usage ?

Reply:
? mainly code from study-pane moved here